docs:#632 single quotes around passwords

micheleRP opened the pull request:
resolves #632 

This adds singles quotes around passwords in case of special characters.  

preview:
- https://deploy-preview-922--redpanda-documentation.netlify.app/docs/platform/security/authentication/
- https://deploy-preview-922--redpanda-documentation.netlify.app/docs/platform/security/authorization/#rpk-for-managing-users-and-acls
- https://deploy-preview-922--redpanda-documentation.netlify.app/docs/platform/security/sasl-kubernetes/
- https://deploy-preview-922--redpanda-documentation.netlify.app/docs/platform/reference/redpanda-operator/kubernetes-sasl/
- https://deploy-preview-922--redpanda-documentation.netlify.app/docs/platform/reference/node-configuration-sample/
- https://deploy-preview-922--redpanda-documentation.netlify.app/docs/platform/console/reference/config/

larsenpanda commented:
I assume these have all been tested, right?

micheleRP commented:
> I assume these have all been tested, right?

@larsenpanda: No, this only makes edits like:
-  -p <password>
+  -p <`password`>

micheleRP commented:
@larsenpanda: Gah, that doesn't render properly. I didn't test all these pages. This PR only puts the text in single quotes; e.g., https://github.com/redpanda-data/documentation/pull/922/files#diff-014c8c5d0bb8eab686b1f3e93762081ffbff3f37735a358c5765531b93fb3474

JakeSCahill commented on docs/platform/reference/node-configuration-sample.mdx:
```diff
@@ -334,7 +335,7 @@ rpk:
     sasl:
       user: user
-      password: pass
+      password: `pass`
```
check all single quotes just in case

micheleRP replied:
> 

gah, thanks @JakeSCahill!

JakeSCahill commented on docs/platform/security/authentication.mdx:
```diff
@@ -265,9 +269,9 @@ schema_registry:
-curl -u "foo:bar" "http://localhost:8082/topics" # A request to the HTTP Proxy with user foo. Don't forget the colon!
-curl -u "foo:bar" "http://localhost:8081/subjects" # A request to the Schema Registry with user foo. Don't forget the colon!
+rpk acl user create foo --password `bar` # Creates SASL user "foo" for the Kafka API
+curl -u "foo:`bar`" "http://localhost:8082/topics" # A request to the HTTP Proxy with user foo. Don't forget the colon!
```
I don't think this is necessary or will even work in cURL commands. @larsenpanda ?

micheleRP replied:
I removed from the curl commands. Thanks @JakeSCahill!

JakeSCahill commented on docs/platform/security/authorization.mdx:
```diff
@@ -326,7 +326,7 @@ Every `rpk acl` command can use these flags:
 <tr>
 <td>--password</td>
-<td>SASL password to be used for authentication.</td>
+<td>SASL password to be used for authentication. Passwords with special characters must be in single quotes.</td>
```
```suggestion
<td>SASL password to be used for authentication. Passwords with special characters must be in single quotes to preserve the literal value of each character.</td>
```

JakeSCahill commented on docs/platform/security/sasl-kubernetes.mdx:
```diff
@@ -25,11 +25,12 @@ auth:
-In this YAML document, `users` is a list of superusers.
+:::note notes
+- In this YAML document, `users` is a list of superusers.
+- Passwords are in single quotes for potential special characters.
```
```suggestion
- Passwords are in single quotes to avoid conflicts with special characters. Enclosing characters in single quotes preserves the literal value of each character.
```

JakeSCahill reviewed (commented):
Some general comments and things to double-check.

larsenpanda commented:
It's single quotes and not backticks, right?

micheleRP commented:
Thank you, @larsenpanda and @JakeSCahill! I think today's soccer messed with my brain. Please review this edited PR.

micheleRP commented:
@larsenpanda can you please approve this, or let me know what else this PR requires?

Feediver1 commented:
@micheleRP I see that this is approved by docs--jut awaiting the SME approval?

BenPope commented on docs/platform/console/reference/config.mdx:
```diff
@@ -65,6 +65,7 @@ kafka:
     enabled: false
     username:
     # Password can also be set using the --kafka.sasl.password flag.
+    # Put passwords in single quotes to avoid conflicts with special characters.
```
I guess this is true of `yaml`, as long as the password doesn't contain a single quote. 

It's true of every string in yaml. The username could contain a single quote, or a "special character".

This seems a separate concern to quoting in bash.

BenPope commented on docs/platform/reference/redpanda-operator/kubernetes-sasl.mdx:
```diff
@@ -140,16 +140,20 @@ kubectl apply -f https://raw.githubusercontent.com/redpanda-data/redpanda-exampl
 You must create the superuser through the Admin API. This user has [ALL permissions](../../../security/authorization#operations) on the cluster and is the user that grants permissions to new users. Without a superuser, you can create other users, but you will not be able to grant them permissions to the cluster.
 
+:::note
+Passwords are in single quotes to avoid conflicts with special characters. Enclosing characters in single quotes preserves the literal value of each character.
```
This is probably true in bash, unless the password contains a single quote. This applies to all strings passed as arguments in bash. The username could contain a single quote. Other shells are available.

BenPope replied:
We might prefer to say something a bit more generic, such "Ensure the arguments are properly quoted". 

That might be less useful;.anybody not using something that's essentially bash is likely to know how their shell does quoting.

BenPope reviewed (commented):
I'm not convinced this really addresses the essence of the problem.

BenPope commented on docs/platform/reference/redpanda-operator/kubernetes-sasl.mdx:
```diff
@@ -193,17 +197,17 @@ The superuser can grant permissions to additional users through access control l
---user <super_user_username> \
---password <super_user_password> \
+--user <superuser_username> \
+--password <'superuser_password'> \
```
This kind of looks like the `<` is required, outside of the quoting. I think `'<superuser_password>'` might be clearer.

micheleRP commented:
@BenPope Thanks so much! I've updated the notes to say either 
To avoid conflicts with special characters, ensure arguments are properly quoted.
or 
Passwords are in single quotes to avoid conflicts with special characters. Ensure arguments are properly quoted.

I've also changed `<'superuser_password'>` to `'<superuser_password>'` and `<'password'>` to `'<password>'`.

BenPope commented on docs/platform/console/reference/config.mdx:
```diff
@@ -45,8 +45,9 @@ The following `config.yaml` configuration file contains a complete list of all C
-This configuration file contains both Redpanda Enterprise and Redpanda Community configurations. If you don't provide a Redpanda Enterprise license, Console ignores configurations for Redpanda Enterprise features.
+:::note notes
+- This configuration file contains both Redpanda Enterprise and Redpanda Community configurations. If you don't provide a Redpanda Enterprise license, Console ignores configurations for Redpanda Enterprise features.
+- To avoid conflicts with special characters, ensure arguments are properly quoted.
```
```suggestion
- Ensure string values are properly quoted if they contain special characters.
```

BenPope commented on docs/platform/reference/redpanda-operator/kubernetes-sasl.mdx:
```diff
@@ -140,16 +140,20 @@ kubectl apply -f https://raw.githubusercontent.com/redpanda-data/redpanda-exampl
 You must create the superuser through the Admin API. This user has [ALL permissions](../../../security/authorization#operations) on the cluster and is the user that grants permissions to new users. Without a superuser, you can create other users, but you will not be able to grant them permissions to the cluster.
 
+:::note
+Passwords are in single quotes to avoid conflicts with special characters. Ensure arguments are properly quoted.
```
```suggestion
Ensure arguments are properly quoted for your shell if they contain special characters.
```

BenPope reviewed (commented):
I think it's worth quoting username, as well.

micheleRP commented:
Copying the edits here into a new PR #1187 in dev.
